File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> set = new HashSet<>();
        for(int num: nums)
            set.add(num);
        int n = nums.length;
        int res = 0;
        for(int i=0;i<n;i++){
            int cur = nums[i];
            if(!set.contains(cur-1)){
                while(set.contains(cur)){
                    cur++;
                }
                res = Math.max(res,cur-nums[i]);
            }
        }
        return res;
    }
}

File: Data Structures & Algorithms/longest-increasing-subsequence/submission-2.cpp
class Solution {
public:
    int lengthOfLIS(vector<int>& nums) {
        int n = nums.size();
        vector<int> ans(n,1);
        if(n==1)
            return 1;
        int maxEle = INT_MIN;
        for(int i=0;i<n-1;i++){
            for(int j=i+1;j<n;j++){
                if(nums[i] < nums[j]){
                    ans[j] = max(ans[i]+1,ans[j]);
                }
                maxEle = max(maxEle,ans[j]);
            }
        }
        return maxEle;
    }
};
